### PR TITLE
Forward retry_on_empty and retries by calling transaction

### PR DESCRIPTION
**Pull request by cgernert**
Fix for https://github.com/pymodbus-dev/pymodbus/issues/1331

### CODE REVIEW

**Comment by janiversen** [comment]
You have formatting issue (black, see CI), you can run ./check_ci.sh locally to have black fix it.

Apart from that all tests fail.

**Comment by cgernert** [comment]
> You have formatting issue (black, see CI), you can run ./check_ci.sh locally to have black fix it.
> 
> Apart from that all tests fail.

OK, fixed it and applied black

File: pymodbus/client/base.py [hunk 1]
```diff
@@ -122,7 +122,9 @@ def __init__(
 
         # Common variables.
         self.framer = self.params.framer(ClientDecoder(), self)
-        self.transaction = DictTransactionManager(self, **kwargs)
+        self.transaction = DictTransactionManager(
+            self, **{**kwargs, **{"retries": retries, "retry_on_empty": retry_on_empty}}
```
Comment (janiversen) [inline pymodbus/client/base.py:126]:
```suggestion
            self, retries=retries, retry_on_empty=retry_on_empty, **kwargs
```

**Review by janiversen — CHANGES_REQUESTED** [review]

**Review by janiversen — APPROVED** [review]
LGTM, thanks.

File: pymodbus/client/base.py (hunk 1 above)
Comment (cgernert) [reply pymodbus/client/base.py]: I did not know you can do that. Looks better than my attempt to merge the dicts.
I am usually writing PHP and Java, so my Python code isn't super clean 😉

**Pull request merged**